modules/sql-engine/src/test/java/org/apache/ignite/internal/sql/engine/planner/datatypes/BaseTypeCoercionTest.java

```diff
@@ -224,4 +228,30 @@ Arguments secondOpBeSame() {
             return Arguments.of(pair, firstOpMatcher, ofTypeWithoutCast(pair.second()));
         }
     }
+
+    /**
+     * Generates SQL literal with a random value for given type.
+     *
+     * @param type Type to generate literal value.
+     * @param auxiliaryType Type which can affect generating value for main type.
+     *
+     * @return Generated value as string representation of a SQL literal.
+     */
+    static String generateLiteral(NativeType type, NativeType auxiliaryType) {
```

Comment:
this `auxiliaryType` is confusing. You don't need another type, but what do you really need is to generate values from upper part of the range (closer to bound). Thus, probably, will be better to change `NativeType auxiliaryType` to `boolean closerToBound` (name on your choice)

Reply:
Seems yes, I like you suggestion